### DISCUSSION TITLE
ops(mempool) sorted

### DISCUSSION THREAD

**Discussion by octavei** [silius-rs/silius#169] [Q&A]
We assume that a person has many ops in the transaction pool, and the op monotonically increases the max_priority_fee_per_gas value. At this time, since the max_priority_fee_per_gas is ranked first, it eventually leads to the op with a large Nonce being submitted first, and the verification simulation fails.

code in project(crates/uopool/src/memory/mempool.rs):
```

    fn get_sorted(&self) -> anyhow::Result<Self::UserOperations> {
        let mut uos: Vec<UserOperation> = self.user_operations.values().cloned().collect();
        uos.sort_by(|a, b| {
            if a.max_priority_fee_per_gas != b.max_priority_fee_per_gas {
                b.max_priority_fee_per_gas.cmp(&a.max_priority_fee_per_gas)
            } else {
                a.nonce.cmp(&b.nonce)
            }
        });
        Ok(uos)
    }
```

**Comment by octavei** [comment]
Nice project. Has anyone already adapted it to a production environment?

**Reply by zsluedem** [reply]
Sadly no. We are looking for people to use it now.

**Reply by octavei** [reply]
There are still some small details that have not been dealt with. But already very good. We'd like to improve it together.

**Reply by zsluedem** [reply]
Are you going to use our bundler? You are more than welcome to use the bundler. Do you want to have a chat together?

**Reply by octavei** [reply]
This is an excellent project, we are currently planning to use it, but what we do may be different from yours. We apply zero-knowledge proof to sign the op under the chain, and Nonce value and Gas (deposit) are stored off-chain.

**Reply by Vid201** [reply]
Cool! You can also fork the code or use only certain crates if your use case differs from standard bundler a lot. :)

**Comment by octavei** [comment]
Like the sorting problem of the transaction pool above, it is actually the same problem as the Nonce value problem I mentioned earlier. At present, each person can only have one op at the same time. If it is changed to be compatible with many ops, the sorting function of the transaction pool also needs to be further changed.

**Reply by zsluedem** [reply]
Yes. I think you totally make sense. I took your issue as a feature request now.We would address that later.

**Comment by Vid201** [comment]
Just adding that there were some changes to 4337 about how the nonces are handled: https://eips.ethereum.org/EIPS/eip-4337#semi-abstracted-nonce-support

Bundler doesn't yet handle that. We are happy to accept contributions @octavei
